src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/service/ClientsService.kt

```diff
@@ -193,8 +194,35 @@ class ClientsService(
     val clientClientConfigPair = retrieveLatestClientWithClientConfig(clientId)
     val client = clientClientConfigPair.first
     val clientConfig = clientClientConfigPair.second
+    val clientDeploymentDetails = getDeployment(clientId)
     setValidDays(clientConfig)
-    return ClientComposite(client, clientConfig, retrieveAuthorizationConsent(client))
+    return ClientComposite(client, clientConfig, retrieveAuthorizationConsent(client), clientDeploymentDetails)
+  }
+  private fun getDeployment(clientId: String): ClientDeploymentDetails {
```

Comment:
This function should just return a Kotlin nullable. Given that it is now being used in the context of the complete client, the deployment may not be present (if the client has been saved in a partially complete state) so no longer needs to throw an exception if the deployment is not present